### PR TITLE
test(native-filters): scoping tree in native filters modal

### PR DESCRIPTION
**Pull request by simcha90**
### SUMMARY

 - Add test to Scoping tree in native filters modal 
 - Includes also tests for FiltersScope functionality in native filters modal

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/spec/fixtures/mockDashboardLayout.js [hunk 1]
```diff
@@ -189,6 +189,49 @@ export const dashboardLayoutWithTabs = {
   future: [],
 };
 
+export const dashboardLayoutWithTabsCombined = {
```
Comment (villebro) [inline superset-frontend/spec/fixtures/mockDashboardLayout.js:192]: Could we have a more descriptive name here: how is this functionally different from `dashboardLayoutWithTabs`?

File: superset-frontend/spec/javascripts/dashboard/components/nativeFilters/FilterScope_spec.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      </Provider>
+    );
+  };
+
+  const getWrapper = (scope?: object) => {
+    render(<MockModal scope={scope} />);
+  };
+
+  const getTreeSwitcher = (order = 0) =>
+    document.querySelectorAll('.ant-tree-switcher')[order];
```
Comment (villebro) [inline superset-frontend/spec/javascripts/dashboard/components/nativeFilters/FilterScope_spec.tsx:63]: Would be nice to add a `data-test` to the Tree switcher.

**Review by villebro — COMMENTED** [review]
A few comments

File: superset-frontend/spec/javascripts/dashboard/components/nativeFilters/FilterScope_spec.tsx (hunk 2 above)
Comment (simchaNielsen) [reply superset-frontend/spec/javascripts/dashboard/components/nativeFilters/FilterScope_spec.tsx:63]: I didn't find way to add `data-test` attribute to the Tree switcher, it looks like Ant did propose such api

Comment (villebro) [reply superset-frontend/spec/javascripts/dashboard/components/nativeFilters/FilterScope_spec.tsx:63]: Oh, you're right, this appears to only be supported on certain AntD components currently.

**Review by villebro — APPROVED** [review]

**Pull request merged**